hazelcast/src/main/java/com/hazelcast/config/AbstractSymmetricEncryptionConfig.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
```

Comment:
Minor - I was planning, as part of next package cleanup, to move all `Abstract*Config` classes to internal as I don't see the user needing to do anything with those. WDYT, should we move this one as well as `SecureStoreConfig`?

Reply:
Hm, is it really a good idea to let public classes/interfaces implement/extend internal classes/interfaces? At the very least, internal APIs are not included in the generated Javadocs (not sure how clever Javadoc is when it comes to 'inheriting' excluded documentation).

Reply:
Talked with Marko, we can take the middle-ground: the abstract classes should be package-private to indicate they are not for public use.